tests/integration/test_charm.py

```diff
@@ -0,0 +1,80 @@
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+log = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+
+APP_NAME = "training-operator"
+CHARM_ROOT = "."
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Build the charm and deploy it with trust=True.
+
+    Assert on the unit status.
+    """
+    charm_under_test = await ops_test.build_charm(CHARM_ROOT)
+    image_path = METADATA["resources"]["training-operator-image"]["upstream-source"]
+    resources = {"training-operator-image": image_path}
+
+    await ops_test.model.deploy(
+        charm_under_test, resources=resources, application_name=APP_NAME, trust=True
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", timeout=60 * 10
+    )
+    assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
+
+
+@pytest.mark.abort_on_fail
+async def test_create_training_jobs(ops_test: OpsTest):
+    """Create training jobs.
+
+    Assert on the ability to create all training jobs from manifests.
+    """
+    # TODO: Kubeflow upstream MXNet examples use GPU.
+    # Not testing MXNetjobs can be created correctly
+    # until we have a CPU mxjob example.
+    examples_dir = Path.cwd() / "examples"
+    kubectl_create = (
+        "microk8s",
+        "kubectl",
+        "create",
+        "-f",
+        examples_dir,
+    )
+
+    ret_code, stdout, stderr = await ops_test.run(*kubectl_create)
+    assert (
+        ret_code == 0
+    ), f"Test returned code {ret_code} with stdout:\n{stdout}\nstderr:\n{stderr}"
+
+
+@pytest.mark.abort_on_fail
+async def test_can_i_create_training_jobs(ops_test: OpsTest):
+    """Validate ability to interact with training jobs with current auth config.
+
+    Asserts whether training-operator can create training jobs with current auth config.
+    """
+    training_jobs = ("tfjobs", "pytorchjobs", "mxjobs", "xgboostjobs")
+    for tjob in training_jobs:
+        kubectl_get = (
+            "microk8s",
```

Comment:
I would prefer if these test could run on other k8s also.

You can check this [PR](https://github.com/canonical/istio-operators/pull/34) from @johnsca for more info.